drainer/collector_test.go

```diff
@@ -463,3 +517,34 @@ func (s *keepUpdatingSuite) TestShouldUpdateWhenNotified(c *C) {
 	s.waitForSignal(c, signal, "Notifies don't finish in time")
 	c.Assert(callCount, Equals, 10)
 }
+
+func (s *keepUpdatingSuite) TestUpdateStatus(c *C) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	etcdClient := etcd.NewClient(testEtcdCluster.RandClient(), DefaultRootPath)
+	r := node.NewEtcdRegistry(etcdClient, time.Duration(5)*time.Second)
+	ns := &node.Status{
+		NodeID:  "test",
+		Addr:    "test",
+		State:   node.Online,
+		IsAlive: true,
+	}
+	err := r.UpdateNode(context.Background(), "pumps", ns)
+	c.Assert(err, IsNil)
+
+	col := Collector{
+		merger:     &Merger{latestTS: 5, sources: map[string]MergeSource{}},
+		notifyChan: make(chan *notifyResult),
+		interval:   10 * time.Second,
+		reg:        r,
+		pumps:      map[string]*Pump{},
+		tiStore:    dummyStorage{},
+	}
+	var TS int64 = 5
+	err = col.updateStatus(ctx)
```

Comment:
You can create a separate suite for test cases of `Collector.updateStatus`. This suite is for test cases of `Collector.keepUpdatingStatus`.